config/logging.yaml

```diff
@@ -0,0 +1,46 @@
+# enable logging output inline in rails console
+:console_inline: true
+# colorize log including level token based on severity
+:colorize: true
+# adds an extra line after each log message with information from which file:line and method
+# was this message logged
+:log_trace: false
+#do not log certain paths (example  /notices/get_new)
+:ignored_paths: []
+# configuration of all loggers used in app
+:loggers:
+  # this logger is parent of all other loggers, by default this is the only one having appender
+  # which is used by all other loggers
+  :root:
+    # only messages with severity >= this level from all loggers will be logged
+    :level: debug
+    # file or syslog, for file variant you can specify other options below like filename,
+    # pattern or rolling options
+    :type: file
+    # 'daily', 'weekly', 'monthly' or number as seconds
+    :age: 'weekly'
+    # how many old log files to keep (156 = 52 * 3 = 3 years for weekly age)
+    :keep: 156
+    # logger line pattern, you can use following macros
+    #  [%c] name of the logger that generate the log event
+    #  [%d] datestamp
+    #  [%m] the user supplied log message
+    #  [%p] PID of the current process
+    #  [%r] the time in milliseconds since the program started
+    #  [%T] the name of the thread Thread.current[:name]
+    #  [%t] object_id of the thread
+    #  [%F] filename where the logging request was issued
+    #  [%L] line number where the logging request was issued
+    #  [%M] method name where the logging request was issued
+    #  [%X{string}] variable set using ::Logging.mdc['string'] =
+    :pattern: "%d [%c] [%.1l] %m\n"
+    # log filename for this environment, it will be placed into #{Rails.root}/log directory
+    :filename: "production.log"
```

Comment:
Testing configuring it to syslog worked (and pointed out that it wasn't logging it as the 'foreman' user properly). Out of the box it looked like:

```
May 13 16:00:25 katello-develnew foreman[21787]: INFO  app : Processing by UsersController#login as HTML
May 13 16:00:25 katello-develnew foreman[21787]: INFO  app :   Rendered users/login.html.erb within layouts/login (76.8ms)
May 13 16:00:25 katello-develnew foreman[21787]: INFO  app :   Rendered layouts/base.html.erb (317.2ms)
```

Discussion worthy for production; however, I am not a syslog master.